// @mcuc/vision/VisionServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for alopay.vision
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VisionClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var vision_api_vision_pb = __importStar(require("./vision_pb"));
var VisionClient = /** @class */ (function () {
    function VisionClient(hostname, credentials, options) {
        this.methodInfoGetIncomeStatement = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetIncomeStatementReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetIncomeStatementReply.deserializeBinary);
        this.methodInfoGetPaymentToday = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetPaymentTodayReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetPaymentTodayReply.deserializeBinary);
        this.methodInfoGetProfitRate = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetProfitRateReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetProfitRateReply.deserializeBinary);
        this.methodInfoGetAllocationTopUpRate = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetAllocationTopUpRateReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetAllocationTopUpRateReply.deserializeBinary);
        this.methodInfoGetAllocationWithdrawRate = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetAllocationWithdrawRateReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetAllocationWithdrawRateReply.deserializeBinary);
        this.methodInfoGetTopPaymentMethod = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetTopPaymentMethodReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetTopPaymentMethodReply.deserializeBinary);
        this.methodInfoGetTopTeller = new grpcWeb.AbstractClientBase.MethodInfo(vision_api_vision_pb.GetTopTellerReply, function (request) {
            return request.serializeBinary();
        }, vision_api_vision_pb.GetTopTellerReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    VisionClient.prototype.getIncomeStatement = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetIncomeStatement', request, metadata || {}, this.methodInfoGetIncomeStatement, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetIncomeStatement', request, metadata || {}, this.methodInfoGetIncomeStatement);
    };
    VisionClient.prototype.getPaymentToday = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetPaymentToday', request, metadata || {}, this.methodInfoGetPaymentToday, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetPaymentToday', request, metadata || {}, this.methodInfoGetPaymentToday);
    };
    VisionClient.prototype.getProfitRate = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetProfitRate', request, metadata || {}, this.methodInfoGetProfitRate, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetProfitRate', request, metadata || {}, this.methodInfoGetProfitRate);
    };
    VisionClient.prototype.getAllocationTopUpRate = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetAllocationTopUpRate', request, metadata || {}, this.methodInfoGetAllocationTopUpRate, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetAllocationTopUpRate', request, metadata || {}, this.methodInfoGetAllocationTopUpRate);
    };
    VisionClient.prototype.getAllocationWithdrawRate = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetAllocationWithdrawRate', request, metadata || {}, this.methodInfoGetAllocationWithdrawRate, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetAllocationWithdrawRate', request, metadata || {}, this.methodInfoGetAllocationWithdrawRate);
    };
    VisionClient.prototype.getTopPaymentMethod = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetTopPaymentMethod', request, metadata || {}, this.methodInfoGetTopPaymentMethod, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetTopPaymentMethod', request, metadata || {}, this.methodInfoGetTopPaymentMethod);
    };
    VisionClient.prototype.getTopTeller = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/alopay.vision.Vision/GetTopTeller', request, metadata || {}, this.methodInfoGetTopTeller, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/alopay.vision.Vision/GetTopTeller', request, metadata || {}, this.methodInfoGetTopTeller);
    };
    return VisionClient;
}());
exports.VisionClient = VisionClient;
