// @mcuc/odin/OdinServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for odin
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OdinClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var odin_api_odin_pb = __importStar(require("./odin_pb"));
var OdinClient = /** @class */ (function () {
    function OdinClient(hostname, credentials, options) {
        this.methodInfoListActionLogs = new grpcWeb.AbstractClientBase.MethodInfo(odin_api_odin_pb.ListActionLogReply, function (request) {
            return request.serializeBinary();
        }, odin_api_odin_pb.ListActionLogReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    OdinClient.prototype.listActionLogs = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/odin.Odin/ListActionLogs', request, metadata || {}, this.methodInfoListActionLogs, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/odin.Odin/ListActionLogs', request, metadata || {}, this.methodInfoListActionLogs);
    };
    return OdinClient;
}());
exports.OdinClient = OdinClient;
