// @mcuc/odin/OdinServiceClientPb.ts
/**
 * @fileoverview gRPC-Web generated client stub for odin
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!


/* eslint-disable */
// @ts-nocheck


import * as grpcWeb from 'grpc-web';

import * as odin_api_odin_pb from './odin_pb';


export class OdinClient {
  client_: grpcWeb.AbstractClientBase;
  hostname_: string;
  credentials_: null | { [index: string]: string; };
  options_: null | { [index: string]: any; };

  constructor (hostname: string,
               credentials?: null | { [index: string]: string; },
               options?: null | { [index: string]: any; }) {
    if (!options) options = {};
    if (!credentials) credentials = {};
    options['format'] = 'text';

    this.client_ = new grpcWeb.GrpcWebClientBase(options);
    this.hostname_ = hostname;
    this.credentials_ = credentials;
    this.options_ = options;
  }

  methodInfoListActionLogs = new grpcWeb.AbstractClientBase.MethodInfo(
    odin_api_odin_pb.ListActionLogReply,
    (request: odin_api_odin_pb.ListActionLogRequest) => {
      return request.serializeBinary();
    },
    odin_api_odin_pb.ListActionLogReply.deserializeBinary
  );

  listActionLogs(
    request: odin_api_odin_pb.ListActionLogRequest,
    metadata: grpcWeb.Metadata | null): Promise<odin_api_odin_pb.ListActionLogReply>;

  listActionLogs(
    request: odin_api_odin_pb.ListActionLogRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: odin_api_odin_pb.ListActionLogReply) => void): grpcWeb.ClientReadableStream<odin_api_odin_pb.ListActionLogReply>;

  listActionLogs(
    request: odin_api_odin_pb.ListActionLogRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: odin_api_odin_pb.ListActionLogReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/odin.Odin/ListActionLogs',
        request,
        metadata || {},
        this.methodInfoListActionLogs,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/odin.Odin/ListActionLogs',
    request,
    metadata || {},
    this.methodInfoListActionLogs);
  }

}
