// @ilt-core/ilt-sidebar/lib/cjs/components/Default/Header/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var core_1 = require("@material-ui/core");
var YorickRoutes_1 = require("../../../YorickRoutes");
var Logo_1 = __importDefault(require("../../icons/ILT/Filled/Logo"));
var ToggleMenu_1 = __importDefault(require("./ToggleMenu"));
var Default_1 = require("../../Layout/Default");
var getConfiguredStyles = function (_a) {
    var _b;
    var appBar = _a.appBar, logoPosition = _a.logoPosition;
    var w = (_b = document.getElementById('ilt-drawer')) === null || _b === void 0 ? void 0 : _b.clientWidth;
    var delta = logoPosition === 'Drawer' ? w : 0;
    return core_1.makeStyles(function (theme) {
        return core_1.createStyles({
            appBar: {
                zIndex: theme.zIndex.drawer + 1,
                backgroundColor: (appBar === null || appBar === void 0 ? void 0 : appBar.background) || '#fff',
                width: "calc(100vw - " + delta + "px)",
                borderBottom: (appBar === null || appBar === void 0 ? void 0 : appBar.border) || '1px solid #ebebeb',
                boxShadow: 'none',
            },
            toolbar: {
                minHeight: (appBar === null || appBar === void 0 ? void 0 : appBar.height) || Default_1.APP_BAR_HEIGHT,
                overflow: 'auto',
            },
            rightGrid: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'flex-end',
                flexGrow: 1,
            },
        });
    });
};
var DefaultHeader = function () {
    var _a, _b, _c, _d, _e;
    var config = react_1.useContext(YorickRoutes_1.TemplateContext).config;
    var classes = getConfiguredStyles(config)();
    var theme = react_1.useMemo(function () { return Default_1.getConfiguredTheme(config); }, []);
    var ComponentLogo = config.logo || react_1.default.createElement(Logo_1.default, null);
    var ComponentProfile = ((_a = config.appBar) === null || _a === void 0 ? void 0 : _a.renderAvatar) || (react_1.default.createElement(core_1.ThemeProvider, { theme: theme },
        react_1.default.createElement(ToggleMenu_1.default, null)));
    return (react_1.default.createElement(core_1.AppBar, { className: classes.appBar + " " + ((_b = config === null || config === void 0 ? void 0 : config.appBar) === null || _b === void 0 ? void 0 : _b.className) },
        react_1.default.createElement(core_1.Toolbar, { className: classes.toolbar },
            config.logoPosition !== 'Drawer' && (react_1.default.createElement(react_router_dom_1.Link, { to: "/" }, ComponentLogo)), (_c = config === null || config === void 0 ? void 0 : config.appBar) === null || _c === void 0 ? void 0 :
            _c.renderAfterLogo,
            react_1.default.createElement("div", { className: classes.rightGrid }, (_d = config === null || config === void 0 ? void 0 : config.appBar) === null || _d === void 0 ? void 0 :
                _d.renderBeforeAvatar,
                ComponentProfile, (_e = config === null || config === void 0 ? void 0 : config.appBar) === null || _e === void 0 ? void 0 :
                _e.renderAfterAvatar))));
};
exports.default = DefaultHeader;
