// @ilt-core/ilt-sidebar/lib/cjs/components/Layout/Default/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useUpdateProfile = exports.getConfiguredTheme = exports.APP_BAR_HEIGHT = exports.DRAWER_WIDTH = void 0;
var react_1 = __importStar(require("react"));
var ScopedCssBaseline_1 = __importDefault(require("@material-ui/core/ScopedCssBaseline"));
var core_1 = require("@material-ui/core");
var YorickRoutes_1 = require("../../../YorickRoutes");
var Header_1 = __importDefault(require("../../Default/Header"));
var Sidebar_1 = __importDefault(require("../../Default/Sidebar"));
var Breadcrumb_1 = __importDefault(require("../../Breadcrumb"));
var ToggleMenu_1 = require("../../Default/Header/ToggleMenu");
exports.DRAWER_WIDTH = 240;
exports.APP_BAR_HEIGHT = 56;
var getConfiguredTheme = function (config) {
    var color = config.color, appBar = config.appBar, drawer = config.drawer;
    return core_1.createMuiTheme({
        palette: {
            primary: {
                main: (color === null || color === void 0 ? void 0 : color.primary) || '#F9912B', // orange
            },
            secondary: {
                main: (color === null || color === void 0 ? void 0 : color.secondary) || '#333E4B',
                light: '#516177',
            },
            warning: {
                main: (color === null || color === void 0 ? void 0 : color.hover) || '#FFF7F0', // light orange
            },
        },
        typography: {
            fontFamily: 'inherit',
            body1: {
                fontFamily: 'inherit',
            },
            body2: {
                fontFamily: 'inherit',
            },
        },
        overrides: {
            MuiToolbar: {
                regular: {
                    minHeight: (appBar === null || appBar === void 0 ? void 0 : appBar.height) || exports.APP_BAR_HEIGHT,
                    '@media(min-width:600px)': {
                        minHeight: (appBar === null || appBar === void 0 ? void 0 : appBar.height) || exports.APP_BAR_HEIGHT,
                    },
                },
            },
            MuiDrawer: {
                root: {
                    minWidth: (drawer === null || drawer === void 0 ? void 0 : drawer.width) || exports.DRAWER_WIDTH,
                    // maxWidth: drawer?.width || DRAWER_WIDTH,
                },
                paper: {
                    background: (drawer === null || drawer === void 0 ? void 0 : drawer.background) || '#fff',
                    padding: '0px 24px',
                    position: 'unset',
                },
                paperAnchorDockedLeft: {
                    borderRight: (drawer === null || drawer === void 0 ? void 0 : drawer.border) || '1px solid #ebebeb',
                },
            },
            // MuiAppBar: {
            //   colorPrimary: {
            //     borderBottom: appBar?.border || '1px solid #ebebeb',
            //     boxShadow: 'none',
            //   },
            // },
            MuiListItemIcon: {
                root: {
                    color: 'inherit',
                    minWidth: 32,
                },
            },
            MuiListItemText: {
                inset: {
                    paddingLeft: 36,
                },
            },
            MuiTypography: {
                body1: {
                    fontSize: 'inherit',
                    fontWeight: 'inherit',
                    color: 'inherit',
                    lineHeight: '20.3px',
                    letterSpacing: 0.75,
                    fontFamily: 'inherit',
                },
            },
            MuiButton: {
                root: {
                    fontFamily: 'inherit',
                },
            },
        },
    });
};
exports.getConfiguredTheme = getConfiguredTheme;
var useStyles = core_1.makeStyles(function () {
    return core_1.createStyles({
        root: {
            display: 'flex',
            width: '100%',
            height: '100%',
        },
        main: {
            flexGrow: 1,
            overflow: 'auto',
        },
    });
});
var useUpdateProfile = function () {
    var updateConfig = react_1.useContext(ToggleMenu_1.ProfileConfig).updateConfig;
    var updateProfile = function (name, avatar) {
        if (updateConfig) {
            updateConfig(name, avatar);
        }
    };
    return { updateProfile: updateProfile };
};
exports.useUpdateProfile = useUpdateProfile;
var DefaultLayout = function (_a) {
    var _b;
    var children = _a.children;
    var classes = useStyles();
    var _c = react_1.useContext(YorickRoutes_1.TemplateContext), config = _c.config, sidebar = _c.sidebar, subSidebar = _c.subSidebar;
    var _d = config.renderBreadcrumbs, renderBreadcrumbs = _d === void 0 ? true : _d;
    var _e = react_1.useState({
        name: '',
        avatar: '',
    }), profile = _e[0], setProfile = _e[1];
    var updateConfig = function (name, avatar) {
        setProfile({ name: name, avatar: avatar || profile.avatar || '' });
    };
    var showMenu = ((_b = config === null || config === void 0 ? void 0 : config.menu) === null || _b === void 0 ? void 0 : _b.show) !== undefined ? config.menu.show : true;
    var theme = react_1.useMemo(function () { return exports.getConfiguredTheme(config); }, []);
    return (react_1.default.createElement(ToggleMenu_1.ProfileConfig.Provider, { value: __assign(__assign({}, profile), { updateConfig: updateConfig }) },
        react_1.default.createElement("div", { className: classes.root },
            react_1.default.createElement(ScopedCssBaseline_1.default, null),
            react_1.default.createElement(Header_1.default, null),
            react_1.default.createElement(core_1.ThemeProvider, { theme: theme }, showMenu && (react_1.default.createElement(Sidebar_1.default, { sidebar: sidebar, subSidebar: subSidebar }))),
            react_1.default.createElement("main", { className: classes.main },
                react_1.default.createElement(core_1.ThemeProvider, { theme: theme },
                    react_1.default.createElement(core_1.Toolbar, null)),
                renderBreadcrumbs && (react_1.default.createElement(Breadcrumb_1.default, { sidebar: sidebar, subSidebar: subSidebar })),
                children))));
};
exports.default = DefaultLayout;
