// @ilt-core/ilt-sidebar/lib/cjs/components/Default/Sidebar/AppMenu.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var core_1 = require("@material-ui/core");
var AppMenuItem_1 = __importDefault(require("./AppMenuItem"));
var YorickRoutes_1 = require("../../../YorickRoutes");
var AllowedTo_1 = __importDefault(require("../../../helper/AllowedTo"));
var useStyles = core_1.makeStyles(function () {
    var drawer = react_1.useContext(YorickRoutes_1.TemplateContext).config.drawer;
    return core_1.createStyles({
        menu: {
            marginTop: 16,
            '& > a': {
                margin: "0px 0px " + ((drawer === null || drawer === void 0 ? void 0 : drawer.spaceSidebar) || 16) + "px",
                '&:last-child': {
                    margin: '0px 0px 0px',
                },
            },
        },
        subMenu: {
            '& > a': {
                margin: "0px 0px " + ((drawer === null || drawer === void 0 ? void 0 : drawer.spaceSubSidebar) || 12) + "px",
            },
        },
        divider: {
            borderTop: (drawer === null || drawer === void 0 ? void 0 : drawer.border) || '1px solid #ebebeb',
            margin: ((drawer === null || drawer === void 0 ? void 0 : drawer.spaceDivider) || 20) + "px 0px",
        },
    });
});
var AppMenu = function (props) {
    var sidebar = props.sidebar, subSidebar = props.subSidebar;
    var classes = useStyles();
    return (react_1.default.createElement(react_1.default.Fragment, null,
        react_1.default.createElement(core_1.List, { component: "nav", disablePadding: true, className: classes.menu }, sidebar === null || sidebar === void 0 ? void 0 : sidebar.map(function (item) { return (react_1.default.createElement(AllowedTo_1.default, { perform: (item === null || item === void 0 ? void 0 : item.permissions) || [], logic: "or" },
            react_1.default.createElement(AppMenuItem_1.default, { item: item, key: item.path }))); })),
        react_1.default.createElement(core_1.List, { component: "nav", disablePadding: true, className: classes.subMenu },
            (subSidebar === null || subSidebar === void 0 ? void 0 : subSidebar.length) ? react_1.default.createElement(core_1.Divider, { className: classes.divider }) : react_1.default.createElement(react_1.default.Fragment, null), subSidebar === null || subSidebar === void 0 ? void 0 :
            subSidebar.map(function (item) { return (react_1.default.createElement(AllowedTo_1.default, { perform: (item === null || item === void 0 ? void 0 : item.permissions) || [], logic: "or" },
                react_1.default.createElement(AppMenuItem_1.default, { item: item, key: item.path, useSub: true }))); }))));
};
exports.default = AppMenu;
