// @ilt-core/ilt-sidebar/lib/esm/components/Default/Sidebar/AppMenuItem.js
import React, { useContext, useMemo, useState } from 'react';
import { List, ListItemIcon, ListItemText, Collapse, useTheme, createStyles, makeStyles, } from '@material-ui/core';
import { useLocation } from 'react-router';
import IconExpandLess from '@material-ui/icons/ExpandLess';
import IconExpandMore from '@material-ui/icons/ExpandMore';
import AppMenuItemComponent from './AppMenuItemComponent';
import { TemplateContext } from '../../../YorickRoutes';
import AllowedTo from '../../../helper/AllowedTo';
var useStyles = makeStyles(function (theme) {
    var drawer = useContext(TemplateContext).config.drawer;
    return createStyles({
        menuItem: {
            width: '100%',
            height: 36,
            borderRadius: 8,
            fontSize: 14,
            fontWeight: 600,
            color: theme.palette.secondary.main,
            '&.active': {
                backgroundColor: theme.palette.warning.main,
                color: theme.palette.primary.main,
                '&::before': {
                    height: '100%',
                    width: '4px',
                    display: 'block',
                    backgroundColor: theme.palette.primary.main,
                    content: '""',
                    borderRadius: '0px 4px 4px 0px',
                    left: '-24px',
                    position: 'absolute',
                },
            },
            '&:hover': {
                background: theme.palette.warning.main,
            },
        },
        menuItemIcon: {
            width: '20px',
            height: '20px',
        },
        // Sub Items
        MuiListRoot: {
            marginTop: 0,
            marginBottom: (drawer === null || drawer === void 0 ? void 0 : drawer.spaceSidebar) || 16,
            '& > a': {
                margin: "0px 0px " + ((drawer === null || drawer === void 0 ? void 0 : drawer.spaceExpandMenu) || 8) + "px",
                fontWeight: 400,
                color: theme.palette.secondary.light,
            },
        },
    });
});
var AppMenuItem = function (props) {
    var classes = useStyles();
    var drawer = useContext(TemplateContext).config.drawer;
    var item = props.item, useSub = props.useSub;
    var title = item.title, path = item.path, icon = item.icon, _a = item.children, children = _a === void 0 ? [] : _a;
    var isExpandable = children && children.length > 0;
    var theme = useTheme();
    var pathname = useLocation().pathname;
    if (pathname.endsWith('/')) {
        pathname = pathname.slice(0, -1);
    }
    var checkCollapseOpen = function (item) {
        var path = item.path, children = item.children;
        if (path.endsWith('/')) {
            path = path.slice(0, -1);
        }
        if (pathname.includes(path))
            return true;
        if (children === null || children === void 0 ? void 0 : children.length) {
            for (var i = 0; i < children.length; i++) {
                if (checkCollapseOpen(children[i]))
                    return true;
            }
        }
        return false;
    };
    var matched = useMemo(function () {
        return checkCollapseOpen(item);
    }, [pathname]);
    var _b = useState(matched), open = _b[0], setOpen = _b[1];
    function handleClick() {
        setOpen(!open);
    }
    var iconStyle = matched
        ? { color: "" + (matched && theme.palette.primary.main) }
        : {};
    var iconExpandLess = (drawer === null || drawer === void 0 ? void 0 : drawer.iconExpandLess) || (React.createElement(IconExpandLess, { className: classes.menuItemIcon }));
    var iconExpandMore = (drawer === null || drawer === void 0 ? void 0 : drawer.iconExpandMore) || (React.createElement(IconExpandMore, { className: classes.menuItemIcon }));
    var MenuItemRoot = (React.createElement(AppMenuItemComponent, { className: classes.menuItem + " ilt-custom-item", link: children.length ? null : path, onClick: handleClick, isActive: pathname.includes(path) },
        !!icon && (React.createElement(ListItemIcon, { style: iconStyle },
            React.createElement(item.icon, null))),
        React.createElement(ListItemText, { primary: title, inset: useSub ? false : !icon }),
        isExpandable && open && iconExpandLess,
        isExpandable && !open && iconExpandMore));
    var MenuItemChildren = isExpandable ? (React.createElement(Collapse, { in: open, timeout: "auto" },
        React.createElement(List, { component: "div", disablePadding: true, className: classes.MuiListRoot }, children === null || children === void 0 ? void 0 : children.map(function (item, index) { return (React.createElement(AllowedTo, { perform: (item === null || item === void 0 ? void 0 : item.permissions) || [], logic: "or" },
            React.createElement(AppMenuItem, { item: item, key: index }))); })))) : null;
    return (React.createElement(React.Fragment, null,
        MenuItemRoot,
        MenuItemChildren));
};
export default AppMenuItem;
