// @mcuc/stark/HowardServiceClientPb.ts
/**
 * @fileoverview gRPC-Web generated client stub for mcuc.stark.howard
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!


/* eslint-disable */
// @ts-nocheck


import * as grpcWeb from 'grpc-web';

import * as stark_api_howard_pb from './howard_pb';


export class HowardClient {
  client_: grpcWeb.AbstractClientBase;
  hostname_: string;
  credentials_: null | { [index: string]: string; };
  options_: null | { [index: string]: any; };

  constructor (hostname: string,
               credentials?: null | { [index: string]: string; },
               options?: null | { [index: string]: any; }) {
    if (!options) options = {};
    if (!credentials) credentials = {};
    options['format'] = 'text';

    this.client_ = new grpcWeb.GrpcWebClientBase(options);
    this.hostname_ = hostname;
    this.credentials_ = credentials;
    this.options_ = options;
  }

  methodInfoGetStatistic = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetStatisticReply,
    (request: stark_api_howard_pb.GetStatisticRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetStatisticReply.deserializeBinary
  );

  getStatistic(
    request: stark_api_howard_pb.GetStatisticRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetStatisticReply>;

  getStatistic(
    request: stark_api_howard_pb.GetStatisticRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetStatisticReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetStatisticReply>;

  getStatistic(
    request: stark_api_howard_pb.GetStatisticRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetStatisticReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetStatistic',
        request,
        metadata || {},
        this.methodInfoGetStatistic,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetStatistic',
    request,
    metadata || {},
    this.methodInfoGetStatistic);
  }

  methodInfoGetProcessingPerformance = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetProcessingPerformanceReply,
    (request: stark_api_howard_pb.GetProcessingPerformanceRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetProcessingPerformanceReply.deserializeBinary
  );

  getProcessingPerformance(
    request: stark_api_howard_pb.GetProcessingPerformanceRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetProcessingPerformanceReply>;

  getProcessingPerformance(
    request: stark_api_howard_pb.GetProcessingPerformanceRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetProcessingPerformanceReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetProcessingPerformanceReply>;

  getProcessingPerformance(
    request: stark_api_howard_pb.GetProcessingPerformanceRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetProcessingPerformanceReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetProcessingPerformance',
        request,
        metadata || {},
        this.methodInfoGetProcessingPerformance,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetProcessingPerformance',
    request,
    metadata || {},
    this.methodInfoGetProcessingPerformance);
  }

  methodInfoGetTotalAmount = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetTotalAmountReply,
    (request: stark_api_howard_pb.GetTotalAmountRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetTotalAmountReply.deserializeBinary
  );

  getTotalAmount(
    request: stark_api_howard_pb.GetTotalAmountRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetTotalAmountReply>;

  getTotalAmount(
    request: stark_api_howard_pb.GetTotalAmountRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTotalAmountReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetTotalAmountReply>;

  getTotalAmount(
    request: stark_api_howard_pb.GetTotalAmountRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTotalAmountReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetTotalAmount',
        request,
        metadata || {},
        this.methodInfoGetTotalAmount,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetTotalAmount',
    request,
    metadata || {},
    this.methodInfoGetTotalAmount);
  }

  methodInfoGetIncomeStatement = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetIncomeStatementReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetIncomeStatementReply.deserializeBinary
  );

  getIncomeStatement(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetIncomeStatementReply>;

  getIncomeStatement(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetIncomeStatementReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetIncomeStatementReply>;

  getIncomeStatement(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetIncomeStatementReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetIncomeStatement',
        request,
        metadata || {},
        this.methodInfoGetIncomeStatement,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetIncomeStatement',
    request,
    metadata || {},
    this.methodInfoGetIncomeStatement);
  }

  methodInfoGetPaymentToday = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetPaymentTodayReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetPaymentTodayReply.deserializeBinary
  );

  getPaymentToday(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetPaymentTodayReply>;

  getPaymentToday(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetPaymentTodayReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetPaymentTodayReply>;

  getPaymentToday(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetPaymentTodayReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetPaymentToday',
        request,
        metadata || {},
        this.methodInfoGetPaymentToday,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetPaymentToday',
    request,
    metadata || {},
    this.methodInfoGetPaymentToday);
  }

  methodInfoGetProfitRate = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetProfitRateReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetProfitRateReply.deserializeBinary
  );

  getProfitRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetProfitRateReply>;

  getProfitRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetProfitRateReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetProfitRateReply>;

  getProfitRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetProfitRateReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetProfitRate',
        request,
        metadata || {},
        this.methodInfoGetProfitRate,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetProfitRate',
    request,
    metadata || {},
    this.methodInfoGetProfitRate);
  }

  methodInfoGetAllocationTopUpRate = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetAllocationTopUpRateReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetAllocationTopUpRateReply.deserializeBinary
  );

  getAllocationTopUpRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetAllocationTopUpRateReply>;

  getAllocationTopUpRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetAllocationTopUpRateReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetAllocationTopUpRateReply>;

  getAllocationTopUpRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetAllocationTopUpRateReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetAllocationTopUpRate',
        request,
        metadata || {},
        this.methodInfoGetAllocationTopUpRate,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetAllocationTopUpRate',
    request,
    metadata || {},
    this.methodInfoGetAllocationTopUpRate);
  }

  methodInfoGetAllocationWithdrawRate = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetAllocationWithdrawRateReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetAllocationWithdrawRateReply.deserializeBinary
  );

  getAllocationWithdrawRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetAllocationWithdrawRateReply>;

  getAllocationWithdrawRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetAllocationWithdrawRateReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetAllocationWithdrawRateReply>;

  getAllocationWithdrawRate(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetAllocationWithdrawRateReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetAllocationWithdrawRate',
        request,
        metadata || {},
        this.methodInfoGetAllocationWithdrawRate,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetAllocationWithdrawRate',
    request,
    metadata || {},
    this.methodInfoGetAllocationWithdrawRate);
  }

  methodInfoGetTopPaymentMethod = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetTopPaymentMethodReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetTopPaymentMethodReply.deserializeBinary
  );

  getTopPaymentMethod(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetTopPaymentMethodReply>;

  getTopPaymentMethod(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTopPaymentMethodReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetTopPaymentMethodReply>;

  getTopPaymentMethod(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTopPaymentMethodReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetTopPaymentMethod',
        request,
        metadata || {},
        this.methodInfoGetTopPaymentMethod,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetTopPaymentMethod',
    request,
    metadata || {},
    this.methodInfoGetTopPaymentMethod);
  }

  methodInfoGetTopTeller = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetTopTellerReply,
    (request: stark_api_howard_pb.GetReportRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetTopTellerReply.deserializeBinary
  );

  getTopTeller(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetTopTellerReply>;

  getTopTeller(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTopTellerReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetTopTellerReply>;

  getTopTeller(
    request: stark_api_howard_pb.GetReportRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetTopTellerReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetTopTeller',
        request,
        metadata || {},
        this.methodInfoGetTopTeller,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetTopTeller',
    request,
    metadata || {},
    this.methodInfoGetTopTeller);
  }

  methodInfoGetSellReportByTimeRange = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetSellReportByTimeRangeReply,
    (request: stark_api_howard_pb.GetSellReportByTimeRangeRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetSellReportByTimeRangeReply.deserializeBinary
  );

  getSellReportByTimeRange(
    request: stark_api_howard_pb.GetSellReportByTimeRangeRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetSellReportByTimeRangeReply>;

  getSellReportByTimeRange(
    request: stark_api_howard_pb.GetSellReportByTimeRangeRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByTimeRangeReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetSellReportByTimeRangeReply>;

  getSellReportByTimeRange(
    request: stark_api_howard_pb.GetSellReportByTimeRangeRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByTimeRangeReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetSellReportByTimeRange',
        request,
        metadata || {},
        this.methodInfoGetSellReportByTimeRange,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetSellReportByTimeRange',
    request,
    metadata || {},
    this.methodInfoGetSellReportByTimeRange);
  }

  methodInfoGetSellReportByMerchant = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetSellReportByMerchantReply,
    (request: stark_api_howard_pb.GetSellReportByMerchantRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetSellReportByMerchantReply.deserializeBinary
  );

  getSellReportByMerchant(
    request: stark_api_howard_pb.GetSellReportByMerchantRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetSellReportByMerchantReply>;

  getSellReportByMerchant(
    request: stark_api_howard_pb.GetSellReportByMerchantRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByMerchantReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetSellReportByMerchantReply>;

  getSellReportByMerchant(
    request: stark_api_howard_pb.GetSellReportByMerchantRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByMerchantReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetSellReportByMerchant',
        request,
        metadata || {},
        this.methodInfoGetSellReportByMerchant,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetSellReportByMerchant',
    request,
    metadata || {},
    this.methodInfoGetSellReportByMerchant);
  }

  methodInfoGetSellReportByPaymentMethod = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetSellReportByPaymentMethodReply,
    (request: stark_api_howard_pb.GetSellReportByPaymentMethodRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetSellReportByPaymentMethodReply.deserializeBinary
  );

  getSellReportByPaymentMethod(
    request: stark_api_howard_pb.GetSellReportByPaymentMethodRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetSellReportByPaymentMethodReply>;

  getSellReportByPaymentMethod(
    request: stark_api_howard_pb.GetSellReportByPaymentMethodRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByPaymentMethodReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetSellReportByPaymentMethodReply>;

  getSellReportByPaymentMethod(
    request: stark_api_howard_pb.GetSellReportByPaymentMethodRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByPaymentMethodReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetSellReportByPaymentMethod',
        request,
        metadata || {},
        this.methodInfoGetSellReportByPaymentMethod,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetSellReportByPaymentMethod',
    request,
    metadata || {},
    this.methodInfoGetSellReportByPaymentMethod);
  }

  methodInfoGetSellReportByTeller = new grpcWeb.AbstractClientBase.MethodInfo(
    stark_api_howard_pb.GetSellReportByTellerReply,
    (request: stark_api_howard_pb.GetSellReportByTellerRequest) => {
      return request.serializeBinary();
    },
    stark_api_howard_pb.GetSellReportByTellerReply.deserializeBinary
  );

  getSellReportByTeller(
    request: stark_api_howard_pb.GetSellReportByTellerRequest,
    metadata: grpcWeb.Metadata | null): Promise<stark_api_howard_pb.GetSellReportByTellerReply>;

  getSellReportByTeller(
    request: stark_api_howard_pb.GetSellReportByTellerRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByTellerReply) => void): grpcWeb.ClientReadableStream<stark_api_howard_pb.GetSellReportByTellerReply>;

  getSellReportByTeller(
    request: stark_api_howard_pb.GetSellReportByTellerRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: stark_api_howard_pb.GetSellReportByTellerReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/mcuc.stark.howard.Howard/GetSellReportByTeller',
        request,
        metadata || {},
        this.methodInfoGetSellReportByTeller,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/mcuc.stark.howard.Howard/GetSellReportByTeller',
    request,
    metadata || {},
    this.methodInfoGetSellReportByTeller);
  }

}
