// @mcuc/stark/PepperServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for mcuc.stark.pepper
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PepperClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var stark_api_pepper_pb = __importStar(require("./pepper_pb"));
var PepperClient = /** @class */ (function () {
    function PepperClient(hostname, credentials, options) {
        this.methodInfoGetBankPaymentCode = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.GetBankPaymentCodeReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.GetBankPaymentCodeReply.deserializeBinary);
        this.methodInfoCreateBankingWithdraw = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.CreateBankingWithdrawReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.CreateBankingWithdrawReply.deserializeBinary);
        this.methodInfoCancelBankingWithdraw = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.CancelBankingWithdrawReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.CancelBankingWithdrawReply.deserializeBinary);
        this.methodInfoCreateBankingTopUp = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.CreateBankingTopUpReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.CreateBankingTopUpReply.deserializeBinary);
        this.methodInfoApproveBankingTopUp = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ApproveBankingTopUpReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ApproveBankingTopUpReply.deserializeBinary);
        this.methodInfoRejectBankingTopUp = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.RejectBankingTopUpReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.RejectBankingTopUpReply.deserializeBinary);
        this.methodInfoListSystemBankAccountByPaymentInfo = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ListSystemBankAccountByPaymentInfoReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ListSystemBankAccountByPaymentInfoReply.deserializeBinary);
        this.methodInfoApproveBankingWithdraw = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ApproveBankingWithdrawReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ApproveBankingWithdrawReply.deserializeBinary);
        this.methodInfoRejectBankingWithdraw = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.RejectBankingWithdrawReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.RejectBankingWithdrawReply.deserializeBinary);
        this.methodInfoSubmitBankingWithdraw = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.SubmitBankingWithdrawReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.SubmitBankingWithdrawReply.deserializeBinary);
        this.methodInfoGetIsMerchantUserBankAccountVerified = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.GetIsMerchantUserBankAccountVerifiedReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.GetIsMerchantUserBankAccountVerifiedReply.deserializeBinary);
        this.methodInfoListSystemBankAccounts = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ListSystemBankAccountsReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ListSystemBankAccountsReply.deserializeBinary);
        this.methodInfoCreateSystemBankAccount = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.CreateSystemBankAccountReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.CreateSystemBankAccountReply.deserializeBinary);
        this.methodInfoVerifyMerchantUserBankAccount = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.VerifyMerchantUserBankAccountReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.VerifyMerchantUserBankAccountReply.deserializeBinary);
        this.methodInfoValidateImportSystemBankAccount = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ValidateImportSystemBankAccountReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ValidateImportSystemBankAccountReply.deserializeBinary);
        this.methodInfoImportSystemBankAccount = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.ImportSystemBankAccountReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.ImportSystemBankAccountReply.deserializeBinary);
        this.methodInfoUpdateSystemBankAccountStatus = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_pepper_pb.UpdateSystemBankAccountStatusReply, function (request) {
            return request.serializeBinary();
        }, stark_api_pepper_pb.UpdateSystemBankAccountStatusReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    PepperClient.prototype.getBankPaymentCode = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/GetBankPaymentCode', request, metadata || {}, this.methodInfoGetBankPaymentCode, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/GetBankPaymentCode', request, metadata || {}, this.methodInfoGetBankPaymentCode);
    };
    PepperClient.prototype.createBankingWithdraw = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/CreateBankingWithdraw', request, metadata || {}, this.methodInfoCreateBankingWithdraw, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/CreateBankingWithdraw', request, metadata || {}, this.methodInfoCreateBankingWithdraw);
    };
    PepperClient.prototype.cancelBankingWithdraw = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/CancelBankingWithdraw', request, metadata || {}, this.methodInfoCancelBankingWithdraw, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/CancelBankingWithdraw', request, metadata || {}, this.methodInfoCancelBankingWithdraw);
    };
    PepperClient.prototype.createBankingTopUp = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/CreateBankingTopUp', request, metadata || {}, this.methodInfoCreateBankingTopUp, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/CreateBankingTopUp', request, metadata || {}, this.methodInfoCreateBankingTopUp);
    };
    PepperClient.prototype.approveBankingTopUp = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ApproveBankingTopUp', request, metadata || {}, this.methodInfoApproveBankingTopUp, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ApproveBankingTopUp', request, metadata || {}, this.methodInfoApproveBankingTopUp);
    };
    PepperClient.prototype.rejectBankingTopUp = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/RejectBankingTopUp', request, metadata || {}, this.methodInfoRejectBankingTopUp, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/RejectBankingTopUp', request, metadata || {}, this.methodInfoRejectBankingTopUp);
    };
    PepperClient.prototype.listSystemBankAccountByPaymentInfo = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ListSystemBankAccountByPaymentInfo', request, metadata || {}, this.methodInfoListSystemBankAccountByPaymentInfo, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ListSystemBankAccountByPaymentInfo', request, metadata || {}, this.methodInfoListSystemBankAccountByPaymentInfo);
    };
    PepperClient.prototype.approveBankingWithdraw = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ApproveBankingWithdraw', request, metadata || {}, this.methodInfoApproveBankingWithdraw, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ApproveBankingWithdraw', request, metadata || {}, this.methodInfoApproveBankingWithdraw);
    };
    PepperClient.prototype.rejectBankingWithdraw = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/RejectBankingWithdraw', request, metadata || {}, this.methodInfoRejectBankingWithdraw, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/RejectBankingWithdraw', request, metadata || {}, this.methodInfoRejectBankingWithdraw);
    };
    PepperClient.prototype.submitBankingWithdraw = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/SubmitBankingWithdraw', request, metadata || {}, this.methodInfoSubmitBankingWithdraw, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/SubmitBankingWithdraw', request, metadata || {}, this.methodInfoSubmitBankingWithdraw);
    };
    PepperClient.prototype.getIsMerchantUserBankAccountVerified = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/GetIsMerchantUserBankAccountVerified', request, metadata || {}, this.methodInfoGetIsMerchantUserBankAccountVerified, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/GetIsMerchantUserBankAccountVerified', request, metadata || {}, this.methodInfoGetIsMerchantUserBankAccountVerified);
    };
    PepperClient.prototype.listSystemBankAccounts = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ListSystemBankAccounts', request, metadata || {}, this.methodInfoListSystemBankAccounts, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ListSystemBankAccounts', request, metadata || {}, this.methodInfoListSystemBankAccounts);
    };
    PepperClient.prototype.createSystemBankAccount = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/CreateSystemBankAccount', request, metadata || {}, this.methodInfoCreateSystemBankAccount, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/CreateSystemBankAccount', request, metadata || {}, this.methodInfoCreateSystemBankAccount);
    };
    PepperClient.prototype.verifyMerchantUserBankAccount = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/VerifyMerchantUserBankAccount', request, metadata || {}, this.methodInfoVerifyMerchantUserBankAccount, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/VerifyMerchantUserBankAccount', request, metadata || {}, this.methodInfoVerifyMerchantUserBankAccount);
    };
    PepperClient.prototype.validateImportSystemBankAccount = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ValidateImportSystemBankAccount', request, metadata || {}, this.methodInfoValidateImportSystemBankAccount, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ValidateImportSystemBankAccount', request, metadata || {}, this.methodInfoValidateImportSystemBankAccount);
    };
    PepperClient.prototype.importSystemBankAccount = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/ImportSystemBankAccount', request, metadata || {}, this.methodInfoImportSystemBankAccount, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/ImportSystemBankAccount', request, metadata || {}, this.methodInfoImportSystemBankAccount);
    };
    PepperClient.prototype.updateSystemBankAccountStatus = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.pepper.Pepper/UpdateSystemBankAccountStatus', request, metadata || {}, this.methodInfoUpdateSystemBankAccountStatus, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.pepper.Pepper/UpdateSystemBankAccountStatus', request, metadata || {}, this.methodInfoUpdateSystemBankAccountStatus);
    };
    return PepperClient;
}());
exports.PepperClient = PepperClient;
