// @mcuc/stark/VisionServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for mcuc.stark.vision
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VisionClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var stark_api_vision_pb = __importStar(require("./vision_pb"));
var VisionClient = /** @class */ (function () {
    function VisionClient(hostname, credentials, options) {
        this.methodInfoGetPaymentDetail = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_vision_pb.GetPaymentDetailReply, function (request) {
            return request.serializeBinary();
        }, stark_api_vision_pb.GetPaymentDetailReply.deserializeBinary);
        this.methodInfoListPayments = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_vision_pb.ListPaymentsReply, function (request) {
            return request.serializeBinary();
        }, stark_api_vision_pb.ListPaymentsReply.deserializeBinary);
        this.methodInfoGetPaymentInfoByPaymentCode = new grpcWeb.AbstractClientBase.MethodInfo(stark_api_vision_pb.GetPaymentInfoByPaymentCodeReply, function (request) {
            return request.serializeBinary();
        }, stark_api_vision_pb.GetPaymentInfoByPaymentCodeReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    VisionClient.prototype.getPaymentDetail = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.vision.Vision/GetPaymentDetail', request, metadata || {}, this.methodInfoGetPaymentDetail, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.vision.Vision/GetPaymentDetail', request, metadata || {}, this.methodInfoGetPaymentDetail);
    };
    VisionClient.prototype.listPayments = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.vision.Vision/ListPayments', request, metadata || {}, this.methodInfoListPayments, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.vision.Vision/ListPayments', request, metadata || {}, this.methodInfoListPayments);
    };
    VisionClient.prototype.getPaymentInfoByPaymentCode = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.stark.vision.Vision/GetPaymentInfoByPaymentCode', request, metadata || {}, this.methodInfoGetPaymentInfoByPaymentCode, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.stark.vision.Vision/GetPaymentInfoByPaymentCode', request, metadata || {}, this.methodInfoGetPaymentInfoByPaymentCode);
    };
    return VisionClient;
}());
exports.VisionClient = VisionClient;
