// @ilt-core/ilt-sidebar/lib/esm/components/LoadingProgressBar/index.js
import React, { createContext, useContext, useEffect, useState } from 'react';
import { createMuiTheme, makeStyles, MuiThemeProvider, } from '@material-ui/core/styles';
import { LinearProgress } from '@material-ui/core';
import { TemplateContext } from '../../YorickRoutes';
var useStyles = makeStyles({
    loadingBar: {
        position: 'absolute',
        zIndex: 9999,
        top: 0,
        width: '100vw',
        backgroundColor: '#fff',
    },
    fullHeight: {
        height: '100vh',
    },
});
var getTheme = function () {
    var color = useContext(TemplateContext).config.color;
    return createMuiTheme({
        palette: {
            primary: {
                main: (color === null || color === void 0 ? void 0 : color.loading) || (color === null || color === void 0 ? void 0 : color.primary) || '#F9930B', // ILT orange
            },
        },
    });
};
export var LoadingBarContext = createContext({
    firstLoad: true,
});
export var LoadingProgressBar = function () {
    var classes = useStyles();
    var theme = getTheme();
    var _a = useState(false), active = _a[0], setActive = _a[1];
    var loadingBarContext = useContext(LoadingBarContext);
    useEffect(function () {
        loadingBarContext.onChange = function (val) {
            setActive(val);
        };
    }, []);
    return (React.createElement(MuiThemeProvider, { theme: theme }, active && (React.createElement("div", { className: classes.loadingBar + " " + (loadingBarContext.firstLoad && classes.fullHeight) },
        React.createElement(LinearProgress, { color: "primary" })))));
};
