// @mcuc/coulson/CoulsonServiceClientPb.ts
/**
 * @fileoverview gRPC-Web generated client stub for coulson
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!


/* eslint-disable */
// @ts-nocheck


import * as grpcWeb from 'grpc-web';

import * as google_protobuf_empty_pb from 'google-protobuf/google/protobuf/empty_pb';
import * as coulson_api_coulson_pb from './coulson_pb';


export class CoulsonClient {
  client_: grpcWeb.AbstractClientBase;
  hostname_: string;
  credentials_: null | { [index: string]: string; };
  options_: null | { [index: string]: any; };

  constructor (hostname: string,
               credentials?: null | { [index: string]: string; },
               options?: null | { [index: string]: any; }) {
    if (!options) options = {};
    if (!credentials) credentials = {};
    options['format'] = 'text';

    this.client_ = new grpcWeb.GrpcWebClientBase(options);
    this.hostname_ = hostname;
    this.credentials_ = credentials;
    this.options_ = options;
  }

  methodInfoGetSetting = new grpcWeb.AbstractClientBase.MethodInfo(
    coulson_api_coulson_pb.GetSettingReply,
    (request: google_protobuf_empty_pb.Empty) => {
      return request.serializeBinary();
    },
    coulson_api_coulson_pb.GetSettingReply.deserializeBinary
  );

  getSetting(
    request: google_protobuf_empty_pb.Empty,
    metadata: grpcWeb.Metadata | null): Promise<coulson_api_coulson_pb.GetSettingReply>;

  getSetting(
    request: google_protobuf_empty_pb.Empty,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: coulson_api_coulson_pb.GetSettingReply) => void): grpcWeb.ClientReadableStream<coulson_api_coulson_pb.GetSettingReply>;

  getSetting(
    request: google_protobuf_empty_pb.Empty,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: coulson_api_coulson_pb.GetSettingReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/coulson.Coulson/GetSetting',
        request,
        metadata || {},
        this.methodInfoGetSetting,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/coulson.Coulson/GetSetting',
    request,
    metadata || {},
    this.methodInfoGetSetting);
  }

  methodInfoEnableAutoApprovalTelco = new grpcWeb.AbstractClientBase.MethodInfo(
    coulson_api_coulson_pb.EnableAutoApprovalTelcoReply,
    (request: coulson_api_coulson_pb.EnableAutoApprovalTelcoRequest) => {
      return request.serializeBinary();
    },
    coulson_api_coulson_pb.EnableAutoApprovalTelcoReply.deserializeBinary
  );

  enableAutoApprovalTelco(
    request: coulson_api_coulson_pb.EnableAutoApprovalTelcoRequest,
    metadata: grpcWeb.Metadata | null): Promise<coulson_api_coulson_pb.EnableAutoApprovalTelcoReply>;

  enableAutoApprovalTelco(
    request: coulson_api_coulson_pb.EnableAutoApprovalTelcoRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: coulson_api_coulson_pb.EnableAutoApprovalTelcoReply) => void): grpcWeb.ClientReadableStream<coulson_api_coulson_pb.EnableAutoApprovalTelcoReply>;

  enableAutoApprovalTelco(
    request: coulson_api_coulson_pb.EnableAutoApprovalTelcoRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: coulson_api_coulson_pb.EnableAutoApprovalTelcoReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/coulson.Coulson/EnableAutoApprovalTelco',
        request,
        metadata || {},
        this.methodInfoEnableAutoApprovalTelco,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/coulson.Coulson/EnableAutoApprovalTelco',
    request,
    metadata || {},
    this.methodInfoEnableAutoApprovalTelco);
  }

}
