// @mcuc/coulson/CoulsonServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for coulson
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoulsonClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var coulson_api_coulson_pb = __importStar(require("./coulson_pb"));
var CoulsonClient = /** @class */ (function () {
    function CoulsonClient(hostname, credentials, options) {
        this.methodInfoGetSetting = new grpcWeb.AbstractClientBase.MethodInfo(coulson_api_coulson_pb.GetSettingReply, function (request) {
            return request.serializeBinary();
        }, coulson_api_coulson_pb.GetSettingReply.deserializeBinary);
        this.methodInfoEnableAutoApprovalTelco = new grpcWeb.AbstractClientBase.MethodInfo(coulson_api_coulson_pb.EnableAutoApprovalTelcoReply, function (request) {
            return request.serializeBinary();
        }, coulson_api_coulson_pb.EnableAutoApprovalTelcoReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    CoulsonClient.prototype.getSetting = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/coulson.Coulson/GetSetting', request, metadata || {}, this.methodInfoGetSetting, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/coulson.Coulson/GetSetting', request, metadata || {}, this.methodInfoGetSetting);
    };
    CoulsonClient.prototype.enableAutoApprovalTelco = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/coulson.Coulson/EnableAutoApprovalTelco', request, metadata || {}, this.methodInfoEnableAutoApprovalTelco, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/coulson.Coulson/EnableAutoApprovalTelco', request, metadata || {}, this.methodInfoEnableAutoApprovalTelco);
    };
    return CoulsonClient;
}());
exports.CoulsonClient = CoulsonClient;
