// @greyhole/recaptcha/RecaptchaServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for recaptcha
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StewardClient = exports.RecaptchaClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var api_recaptcha_pb = __importStar(require("./recaptcha_pb"));
var RecaptchaClient = /** @class */ (function () {
    function RecaptchaClient(hostname, credentials, options) {
        this.methodInfoGetCaptcha = new grpcWeb.AbstractClientBase.MethodInfo(api_recaptcha_pb.GetCaptchaReply, function (request) {
            return request.serializeBinary();
        }, api_recaptcha_pb.GetCaptchaReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    RecaptchaClient.prototype.getCaptcha = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/recaptcha.Recaptcha/GetCaptcha', request, metadata || {}, this.methodInfoGetCaptcha, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/recaptcha.Recaptcha/GetCaptcha', request, metadata || {}, this.methodInfoGetCaptcha);
    };
    return RecaptchaClient;
}());
exports.RecaptchaClient = RecaptchaClient;
var StewardClient = /** @class */ (function () {
    function StewardClient(hostname, credentials, options) {
        this.methodInfoVerifyCaptcha = new grpcWeb.AbstractClientBase.MethodInfo(api_recaptcha_pb.VerifyCaptchaReply, function (request) {
            return request.serializeBinary();
        }, api_recaptcha_pb.VerifyCaptchaReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    StewardClient.prototype.verifyCaptcha = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/recaptcha.Steward/VerifyCaptcha', request, metadata || {}, this.methodInfoVerifyCaptcha, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/recaptcha.Steward/VerifyCaptcha', request, metadata || {}, this.methodInfoVerifyCaptcha);
    };
    return StewardClient;
}());
exports.StewardClient = StewardClient;
