// @greyhole/recaptcha/RecaptchaServiceClientPb.ts
/**
 * @fileoverview gRPC-Web generated client stub for recaptcha
 * @enhanceable
 * @public
 */

// GENERATED CODE -- DO NOT EDIT!


/* eslint-disable */
// @ts-nocheck


import * as grpcWeb from 'grpc-web';

import * as api_recaptcha_pb from './recaptcha_pb';


export class RecaptchaClient {
  client_: grpcWeb.AbstractClientBase;
  hostname_: string;
  credentials_: null | { [index: string]: string; };
  options_: null | { [index: string]: any; };

  constructor (hostname: string,
               credentials?: null | { [index: string]: string; },
               options?: null | { [index: string]: any; }) {
    if (!options) options = {};
    if (!credentials) credentials = {};
    options['format'] = 'text';

    this.client_ = new grpcWeb.GrpcWebClientBase(options);
    this.hostname_ = hostname;
    this.credentials_ = credentials;
    this.options_ = options;
  }

  methodInfoGetCaptcha = new grpcWeb.AbstractClientBase.MethodInfo(
    api_recaptcha_pb.GetCaptchaReply,
    (request: api_recaptcha_pb.GetCaptchaRequest) => {
      return request.serializeBinary();
    },
    api_recaptcha_pb.GetCaptchaReply.deserializeBinary
  );

  getCaptcha(
    request: api_recaptcha_pb.GetCaptchaRequest,
    metadata: grpcWeb.Metadata | null): Promise<api_recaptcha_pb.GetCaptchaReply>;

  getCaptcha(
    request: api_recaptcha_pb.GetCaptchaRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: api_recaptcha_pb.GetCaptchaReply) => void): grpcWeb.ClientReadableStream<api_recaptcha_pb.GetCaptchaReply>;

  getCaptcha(
    request: api_recaptcha_pb.GetCaptchaRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: api_recaptcha_pb.GetCaptchaReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/recaptcha.Recaptcha/GetCaptcha',
        request,
        metadata || {},
        this.methodInfoGetCaptcha,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/recaptcha.Recaptcha/GetCaptcha',
    request,
    metadata || {},
    this.methodInfoGetCaptcha);
  }

}

export class StewardClient {
  client_: grpcWeb.AbstractClientBase;
  hostname_: string;
  credentials_: null | { [index: string]: string; };
  options_: null | { [index: string]: any; };

  constructor (hostname: string,
               credentials?: null | { [index: string]: string; },
               options?: null | { [index: string]: any; }) {
    if (!options) options = {};
    if (!credentials) credentials = {};
    options['format'] = 'text';

    this.client_ = new grpcWeb.GrpcWebClientBase(options);
    this.hostname_ = hostname;
    this.credentials_ = credentials;
    this.options_ = options;
  }

  methodInfoVerifyCaptcha = new grpcWeb.AbstractClientBase.MethodInfo(
    api_recaptcha_pb.VerifyCaptchaReply,
    (request: api_recaptcha_pb.VerifyCaptchaRequest) => {
      return request.serializeBinary();
    },
    api_recaptcha_pb.VerifyCaptchaReply.deserializeBinary
  );

  verifyCaptcha(
    request: api_recaptcha_pb.VerifyCaptchaRequest,
    metadata: grpcWeb.Metadata | null): Promise<api_recaptcha_pb.VerifyCaptchaReply>;

  verifyCaptcha(
    request: api_recaptcha_pb.VerifyCaptchaRequest,
    metadata: grpcWeb.Metadata | null,
    callback: (err: grpcWeb.Error,
               response: api_recaptcha_pb.VerifyCaptchaReply) => void): grpcWeb.ClientReadableStream<api_recaptcha_pb.VerifyCaptchaReply>;

  verifyCaptcha(
    request: api_recaptcha_pb.VerifyCaptchaRequest,
    metadata: grpcWeb.Metadata | null,
    callback?: (err: grpcWeb.Error,
               response: api_recaptcha_pb.VerifyCaptchaReply) => void) {
    if (callback !== undefined) {
      return this.client_.rpcCall(
        this.hostname_ +
          '/recaptcha.Steward/VerifyCaptcha',
        request,
        metadata || {},
        this.methodInfoVerifyCaptcha,
        callback);
    }
    return this.client_.unaryCall(
    this.hostname_ +
      '/recaptcha.Steward/VerifyCaptcha',
    request,
    metadata || {},
    this.methodInfoVerifyCaptcha);
  }

}
