// @ilt-core/ilt-sidebar/lib/cjs/components/Breadcrumb/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_router_dom_1 = require("react-router-dom");
var core_1 = require("@material-ui/core");
var Link_1 = __importDefault(require("@material-ui/core/Link"));
var Typography_1 = __importDefault(require("@material-ui/core/Typography"));
var YorickRoutes_1 = require("../../YorickRoutes");
var RightArrow_1 = __importDefault(require("../icons/ILT/Stroke/RightArrow"));
var LinkRouter = function (props) { return react_1.default.createElement(Link_1.default, __assign({}, props, { component: react_router_dom_1.Link })); };
var getSidebarMapingName = function (sidebar) {
    var res = {};
    sidebar === null || sidebar === void 0 ? void 0 : sidebar.map(function (_a) {
        var path = _a.path, title = _a.title, children = _a.children;
        res[path] = title;
        if (children === null || children === void 0 ? void 0 : children.length) {
            for (var _i = 0, _b = Object.entries(getSidebarMapingName(children)); _i < _b.length; _i++) {
                var _c = _b[_i], key = _c[0], value = _c[1];
                res[key] = value;
            }
        }
    });
    return res;
};
var DefaultBreadcrumbs = function (_a) {
    var sidebar = _a.sidebar, subSidebar = _a.subSidebar;
    var classes = useStyles();
    var pathname = react_router_dom_1.useLocation().pathname;
    var breadcrumbsCfg = react_1.useContext(YorickRoutes_1.TemplateContext).config.breadcrumbs;
    var breadcrumbs = pathname.split('/').filter(function (x) { return x; });
    var mappingName = react_1.useMemo(function () { return (__assign(__assign({}, getSidebarMapingName(sidebar)), getSidebarMapingName(subSidebar))); }, [sidebar, subSidebar]);
    return (react_1.default.createElement(core_1.Breadcrumbs, { "aria-label": "breadcrumb", className: classes.root, separator: react_1.default.createElement(RightArrow_1.default, null) },
        react_1.default.createElement(LinkRouter, { color: "inherit", to: "/", className: classes.normal }, (breadcrumbsCfg === null || breadcrumbsCfg === void 0 ? void 0 : breadcrumbsCfg.homeText) || 'Home'), breadcrumbs === null || breadcrumbs === void 0 ? void 0 :
        breadcrumbs.map(function (_, index) {
            // console.log('path', _);
            var last = index === breadcrumbs.length - 1;
            var to = "/" + breadcrumbs.slice(0, index + 1).join('/');
            var displayName = mappingName[to] || '';
            return last ? (react_1.default.createElement(Typography_1.default, { key: to, className: classes.last }, displayName)) : (react_1.default.createElement(LinkRouter, { to: to, key: to, className: classes.normal }, displayName));
        })));
};
exports.default = DefaultBreadcrumbs;
var useStyles = core_1.makeStyles(function () {
    var breadcrumbs = react_1.useContext(YorickRoutes_1.TemplateContext).config.breadcrumbs;
    return core_1.createStyles({
        root: {
            padding: (breadcrumbs === null || breadcrumbs === void 0 ? void 0 : breadcrumbs.padding) || '32px 0 0 32px',
            '& .MuiBreadcrumbs-separator': {
                margin: '0px 24px',
            },
        },
        normal: {
            fontSize: '14px',
            fontWeight: 600,
            color: '#333E4B',
            lineHeight: '146%',
        },
        last: {
            fontSize: '14px',
            fontWeight: 600,
            color: '#1789FC',
            lineHeight: '146%',
        },
    });
});
