// @ilt-core/ilt-form/lib/cjs/components/Inputs/FormPhoneNumber/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormPhoneNumber = void 0;
var react_1 = __importStar(require("react"));
var core_1 = require("@material-ui/core");
var react_hook_form_1 = require("react-hook-form");
var FormHelperText_1 = __importDefault(require("../../helper/FormHelperText"));
var icons_1 = require("../../icons");
exports.FormPhoneNumber = react_1.default.forwardRef(function (props, ref) {
    var _a;
    var _b = react_hook_form_1.useFormContext(), setValue = _b.setValue, getValues = _b.getValues, errors = _b.formState.errors;
    var register = react_hook_form_1.useFormContext().register;
    var name = props.name, label = props.label, placeholder = props.placeholder, _c = props.value, value = _c === void 0 ? {} : _c, options = props.options, 
    // helperText,
    disabled = props.disabled, 
    // onChange,
    onBlur = props.onBlur, onSync = props.onSync;
    var classes = useStyles();
    var _d = react_1.useState(value.selectValue || '+84'), selectValue = _d[0], setSelectValue = _d[1];
    var _e = react_1.useState(value.inputValue || ''), inputValue = _e[0], setInputValue = _e[1];
    var _f = react_1.useState(false), open = _f[0], setOpen = _f[1];
    var anchorRef = react_1.useRef(null);
    var handleToggle = function () {
        setOpen(function (prevOpen) { return !prevOpen; });
    };
    var handleClose = function (event) {
        if (anchorRef.current &&
            anchorRef.current.contains(event.target)) {
            return;
        }
        setOpen(false);
    };
    function handleListKeyDown(event) {
        if (event.key === 'Tab') {
            event.preventDefault();
            setOpen(false);
        }
    }
    // return focus to the button when we transitioned from !open -> open
    var prevOpen = react_1.default.useRef(open);
    react_1.default.useEffect(function () {
        if (prevOpen.current === true && open === false) {
            anchorRef.current.focus();
        }
        prevOpen.current = open;
    }, [open]);
    var handleInputChange = function (event) {
        var val = event.target.value;
        var newValue = { selectValue: selectValue, inputValue: val };
        setValue(name, newValue);
        setInputValue(val);
        onSync();
    };
    var handleSelectChange = function (value) {
        var newValue = { selectValue: value, inputValue: inputValue };
        setValue(name, newValue, { shouldValidate: false });
        setSelectValue(value);
        setOpen(false);
        onSync();
    };
    var displayFlag = options === null || options === void 0 ? void 0 : options.find(function (item) { return item.value == selectValue; });
    console.log(errors[name]);
    console.log(getValues());
    return (react_1.default.createElement(core_1.FormControl, { fullWidth: true, component: "div", error: !!errors[name], disabled: disabled, onBlur: onBlur },
        react_1.default.createElement(core_1.InputLabel, { shrink: true }, label),
        react_1.default.createElement(core_1.Popper, { open: open, anchorEl: anchorRef.current, role: '', transition: true, placement: "bottom-end", modifiers: {
                offset: {
                    enabled: true,
                    offset: '0, 8', // https://popper.js.org/docs/v1/#modifiersoffset
                },
            }, style: { zIndex: 9999 } }, function (_a) {
            var _b;
            var TransitionProps = _a.TransitionProps, placement = _a.placement;
            return (react_1.default.createElement(core_1.Grow, __assign({}, TransitionProps, { style: {
                    transformOrigin: placement === 'bottom' ? 'top-end' : 'bottom-end',
                    minWidth: (_b = anchorRef.current) === null || _b === void 0 ? void 0 : _b.offsetWidth,
                } }),
                react_1.default.createElement(core_1.Paper, { className: classes.paper },
                    react_1.default.createElement(core_1.ClickAwayListener, { onClickAway: handleClose },
                        react_1.default.createElement(core_1.MenuList, { id: "menu-list-grow", variant: "selectedMenu", autoFocusItem: open, onKeyDown: handleListKeyDown }, (options === null || options === void 0 ? void 0 : options.length) &&
                            options.map(function (_a, index) {
                                var label = _a.label, icon = _a.icon, value = _a.value;
                                return (react_1.default.createElement(core_1.MenuItem, { onClick: function () { return handleSelectChange(value); }, key: index },
                                    react_1.default.createElement(core_1.ListItemIcon, { className: classes.listItemIcon }, icon),
                                    react_1.default.createElement(core_1.Typography, null,
                                        react_1.default.createElement("span", null, label),
                                        "\u00A0",
                                        react_1.default.createElement("span", { className: classes.listItemValue }, value))));
                            }))))));
        }),
        react_1.default.createElement(core_1.InputBase, __assign({}, register(name, {
            required: 'This field is required',
            minLength: { value: 5, message: 'min length 5' },
        }), { fullWidth: true, value: inputValue, inputRef: ref, ref: anchorRef, placeholder: placeholder, onChange: handleInputChange, className: classes.inputBase, inputProps: {
                type: 'number',
                // ...register(name, {
                //   required: 'This field is required',
                //   minLength: { value: 5, message: 'min length 5' },
                // })
            }, startAdornment: react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(core_1.Button, { startIcon: displayFlag === null || displayFlag === void 0 ? void 0 : displayFlag.icon, endIcon: react_1.default.createElement(icons_1.ArrowDropDown, null), className: classes.selectButton, classes: {
                        startIcon: classes.startIcon,
                        endIcon: classes.endIcon,
                    }, onClick: handleToggle }),
                react_1.default.createElement(core_1.Typography, { className: classes.displaySelectValue },
                    "(",
                    selectValue,
                    ")")) })),
        react_1.default.createElement(core_1.FormHelperText, null,
            react_1.default.createElement(FormHelperText_1.default, { message: (_a = errors[name]) === null || _a === void 0 ? void 0 : _a.message }))));
});
var useStyles = core_1.makeStyles({
    selectButton: {
        minWidth: 48,
        color: '#67677A',
        margin: '0px 8px 0px 12px',
        '&:hover': {
            backgroundColor: 'transparent',
        },
    },
    displaySelectValue: {
        width: 45,
        fontStyle: 'normal',
        fontWeight: 'normal',
        fontSize: 16,
        letterSpacing: 0.5,
        // color: '#CCCCCC',
    },
    inputBase: {
        // hide default button increase/decrease
        '& input[type=number]::-webkit-inner-spin-button, & input[type=number]::-webkit-outer-spin-button': {
            '-webkit-appearance': 'none',
            margin: 0,
        },
    },
    listItemIcon: {
        minWidth: 32,
    },
    listItemValue: {
        opacity: 0.5,
    },
    startIcon: {
        marginLeft: -12,
    },
    endIcon: {
        marginLeft: -2,
    },
    paper: {
        border: '1px solid #F5F5F5',
        boxSizing: 'border-box',
        boxShadow: '0px 8px 16px rgba(37, 37, 51, 0.24)',
    },
});
