// @ilt-core/ilt-form/lib/cjs/components/Inputs/FormCheckBoxGroup/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormCheckBoxGroup = void 0;
var react_1 = __importDefault(require("react"));
var core_1 = require("@material-ui/core");
var FormHelperText_1 = __importDefault(require("../../helper/FormHelperText"));
var react_hook_form_1 = require("react-hook-form");
var CheckboxGroup_1 = require("./CheckboxGroup");
var FormCheckBoxGroup = function (props) {
    var _a;
    var name = props.name, label = props.label;
    var errors = react_hook_form_1.useFormContext().formState.errors;
    var classes = useStyles();
    return (react_1.default.createElement(core_1.FormControl, { classes: { root: classes.MuiFormControlRoot } },
        react_1.default.createElement(core_1.FormControlLabel, { classes: { root: classes.MuiFormControlLabel }, label: label, control: react_1.default.createElement(CheckboxGroup_1.CheckboxGroup, __assign({}, props)) }),
        react_1.default.createElement(core_1.FormHelperText, null,
            react_1.default.createElement(FormHelperText_1.default, { message: (_a = errors[name]) === null || _a === void 0 ? void 0 : _a.message }))));
};
exports.FormCheckBoxGroup = FormCheckBoxGroup;
var useStyles = core_1.makeStyles({
    MuiFormControlRoot: {
        marginTop: 0,
        width: '100%',
    },
    MuiFormControlLabel: {
        margin: 0,
    },
});
