// @ilt-core/ilt-sidebar/lib/cjs/YorickRoutes/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.YorickTemplate = exports.TemplateContext = void 0;
var react_1 = __importDefault(require("react"));
var react_router_dom_1 = require("react-router-dom");
var core_1 = require("@material-ui/core");
var CustomRoute_1 = __importDefault(require("./CustomRoute"));
var Default_1 = __importDefault(require("../components/Layout/Default"));
var LoadingProgressBar_1 = require("../components/LoadingProgressBar");
exports.TemplateContext = react_1.default.createContext({
    routes: [],
    sidebar: [],
    subSidebar: [],
    config: {},
    userPermissions: [],
});
var useStyles = core_1.makeStyles({
    mainComponent: {
        zIndex: 10,
        width: '100%',
        height: '100%',
    },
});
var YorickTemplate = function (props) {
    var routes = props.routes;
    var classes = useStyles();
    var NestedRoute = function (routes, firstLayer) { return (react_1.default.createElement(react_router_dom_1.Switch, null, routes === null || routes === void 0 ? void 0 : routes.map(function (route, i) {
        var _a;
        var LayoutComponent = route.layout || Default_1.default;
        return ((_a = route.routes) === null || _a === void 0 ? void 0 : _a.length) ? (react_1.default.createElement(react_router_dom_1.Route, { path: route.path, exact: route.exact, key: i }, firstLayer ? (react_1.default.createElement(LayoutComponent, null, NestedRoute(route.routes))) : (NestedRoute(route.routes)))) : firstLayer ? (react_1.default.createElement(react_router_dom_1.Route, { path: route.path, exact: route.exact, key: i },
            react_1.default.createElement(LayoutComponent, null,
                react_1.default.createElement(react_router_dom_1.Route, { path: route.path, exact: !!route.exact, render: function (routeProps) { return (react_1.default.createElement(CustomRoute_1.default, { route: route, routeProps: routeProps })); } })))) : (react_1.default.createElement(react_router_dom_1.Route, { key: i, path: route.path, exact: !!route.exact, render: function (routeProps) { return (react_1.default.createElement(CustomRoute_1.default, { route: route, routeProps: routeProps })); } }));
    }))); };
    return (react_1.default.createElement(react_router_dom_1.BrowserRouter, null,
        react_1.default.createElement(exports.TemplateContext.Provider, { value: props },
            react_1.default.createElement(LoadingProgressBar_1.LoadingProgressBar, null),
            react_1.default.createElement("div", { className: classes.mainComponent }, NestedRoute(routes, true)))));
};
exports.YorickTemplate = YorickTemplate;
