// @ilt-core/ilt-sidebar/lib/esm/components/Default/Header/index.js
import React, { useContext, useMemo } from 'react';
import { Link } from 'react-router-dom';
import { AppBar, createStyles, makeStyles, ThemeProvider, Toolbar, } from '@material-ui/core';
import { TemplateContext } from '../../../YorickRoutes';
import ILTLogo from '../../icons/ILT/Filled/Logo';
import MenuListComposition from './ToggleMenu';
import { APP_BAR_HEIGHT, getConfiguredTheme } from '../../Layout/Default';
var getConfiguredStyles = function (_a) {
    var _b;
    var appBar = _a.appBar, logoPosition = _a.logoPosition;
    var w = (_b = document.getElementById('ilt-drawer')) === null || _b === void 0 ? void 0 : _b.clientWidth;
    var delta = logoPosition === 'Drawer' ? w : 0;
    return makeStyles(function (theme) {
        return createStyles({
            appBar: {
                zIndex: theme.zIndex.drawer + 1,
                backgroundColor: (appBar === null || appBar === void 0 ? void 0 : appBar.background) || '#fff',
                width: "calc(100vw - " + delta + "px)",
                borderBottom: (appBar === null || appBar === void 0 ? void 0 : appBar.border) || '1px solid #ebebeb',
                boxShadow: 'none',
            },
            toolbar: {
                minHeight: (appBar === null || appBar === void 0 ? void 0 : appBar.height) || APP_BAR_HEIGHT,
                overflow: 'auto',
            },
            rightGrid: {
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'flex-end',
                flexGrow: 1,
            },
        });
    });
};
var DefaultHeader = function () {
    var _a, _b, _c, _d, _e;
    var config = useContext(TemplateContext).config;
    var classes = getConfiguredStyles(config)();
    var theme = useMemo(function () { return getConfiguredTheme(config); }, []);
    var ComponentLogo = config.logo || React.createElement(ILTLogo, null);
    var ComponentProfile = ((_a = config.appBar) === null || _a === void 0 ? void 0 : _a.renderAvatar) || (React.createElement(ThemeProvider, { theme: theme },
        React.createElement(MenuListComposition, null)));
    return (React.createElement(AppBar, { className: classes.appBar + " " + ((_b = config === null || config === void 0 ? void 0 : config.appBar) === null || _b === void 0 ? void 0 : _b.className) },
        React.createElement(Toolbar, { className: classes.toolbar },
            config.logoPosition !== 'Drawer' && (React.createElement(Link, { to: "/" }, ComponentLogo)), (_c = config === null || config === void 0 ? void 0 : config.appBar) === null || _c === void 0 ? void 0 :
            _c.renderAfterLogo,
            React.createElement("div", { className: classes.rightGrid }, (_d = config === null || config === void 0 ? void 0 : config.appBar) === null || _d === void 0 ? void 0 :
                _d.renderBeforeAvatar,
                ComponentProfile, (_e = config === null || config === void 0 ? void 0 : config.appBar) === null || _e === void 0 ? void 0 :
                _e.renderAfterAvatar))));
};
export default DefaultHeader;
