// @ilt-core/ilt-form/lib/esm/components/Inputs/FormAutoComplete/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { makeStyles, TextField } from '@material-ui/core';
import { useFormContext } from 'react-hook-form';
import { Autocomplete } from '@material-ui/lab';
import { Close, Search } from '../../icons';
import MyHelperText from '../../helper/FormHelperText';
var useStyles = makeStyles({
    root: {},
    inputRoot: {
        paddingRight: '1px !important',
        paddingBottom: '0px !important',
        '& $input': {
            padding: '12px !important',
        },
    },
    endAdornment: {
        top: 0,
        position: 'unset',
    },
    clearIndicator: {
        padding: '11px',
        margin: '0px',
    },
    popupIndicator: {
        padding: '11px',
        margin: '0px',
    },
    popupIndicatorOpen: {
        transform: 'rotate(0deg)',
    },
});
export var FormAutoComplete = React.forwardRef(function (props, ref) {
    var classes = useStyles();
    var _a = useFormContext(), setValue = _a.setValue, errors = _a.formState.errors;
    var name = props.name, label = props.label, placeholder = props.placeholder, value = props.value, options = props.options, 
    // helperText,
    disabled = props.disabled, popupIcon = props.popupIcon, 
    // onChange,
    onBlur = props.onBlur, onSync = props.onSync, pure = props.pure, onChangePure = props.onChangePure;
    return (React.createElement(Autocomplete, { fullWidth: true, autoComplete: true, autoHighlight: true, value: (options && options.find(function (option) { return option.value == value; })) || null, classes: classes, options: options || [], groupBy: function (_a) {
            var category = _a.category;
            return category || '';
        }, getOptionLabel: function (_a) {
            var label = _a.label;
            return label;
        }, getOptionDisabled: function (_a) {
            var disabled = _a.disabled;
            return !!disabled;
        }, disabled: !!disabled, onChange: function (e, option) {
            if (pure) {
                onChangePure === null || onChangePure === void 0 ? void 0 : onChangePure(e, option);
            }
            else {
                e.preventDefault();
                setValue(name, option === null || option === void 0 ? void 0 : option.value);
                onSync();
            }
        }, onBlur: onBlur, popupIcon: popupIcon || React.createElement(Search, null), closeIcon: React.createElement(Close, null), renderInput: function (params) {
            var _a;
            return (React.createElement(TextField, __assign({}, params, { label: label, placeholder: placeholder, inputRef: ref, InputLabelProps: {
                    shrink: true,
                }, error: !!errors[name], helperText: React.createElement(MyHelperText, { message: (_a = errors[name]) === null || _a === void 0 ? void 0 : _a.message }) })));
        } }));
});
