// @mcuc/groot/GrootServiceClientPb.js
"use strict";
/**
 * @fileoverview gRPC-Web generated client stub for mcuc.groot
 * @enhanceable
 * @public
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GrootClient = void 0;
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
var grpcWeb = __importStar(require("grpc-web"));
var groot_api_groot_pb = __importStar(require("./groot_pb"));
var GrootClient = /** @class */ (function () {
    function GrootClient(hostname, credentials, options) {
        this.methodInfoGetCard = new grpcWeb.AbstractClientBase.MethodInfo(groot_api_groot_pb.Card, function (request) {
            return request.serializeBinary();
        }, groot_api_groot_pb.Card.deserializeBinary);
        this.methodInfoChargeCard = new grpcWeb.AbstractClientBase.MethodInfo(groot_api_groot_pb.ChargeCardReply, function (request) {
            return request.serializeBinary();
        }, groot_api_groot_pb.ChargeCardReply.deserializeBinary);
        this.methodInfoGetTxIDByCard = new grpcWeb.AbstractClientBase.MethodInfo(groot_api_groot_pb.GetTxIDByCardReply, function (request) {
            return request.serializeBinary();
        }, groot_api_groot_pb.GetTxIDByCardReply.deserializeBinary);
        if (!options)
            options = {};
        if (!credentials)
            credentials = {};
        options['format'] = 'text';
        this.client_ = new grpcWeb.GrpcWebClientBase(options);
        this.hostname_ = hostname;
        this.credentials_ = credentials;
        this.options_ = options;
    }
    GrootClient.prototype.getCard = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.groot.Groot/GetCard', request, metadata || {}, this.methodInfoGetCard, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.groot.Groot/GetCard', request, metadata || {}, this.methodInfoGetCard);
    };
    GrootClient.prototype.chargeCard = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.groot.Groot/ChargeCard', request, metadata || {}, this.methodInfoChargeCard, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.groot.Groot/ChargeCard', request, metadata || {}, this.methodInfoChargeCard);
    };
    GrootClient.prototype.getTxIDByCard = function (request, metadata, callback) {
        if (callback !== undefined) {
            return this.client_.rpcCall(this.hostname_ +
                '/mcuc.groot.Groot/GetTxIDByCard', request, metadata || {}, this.methodInfoGetTxIDByCard, callback);
        }
        return this.client_.unaryCall(this.hostname_ +
            '/mcuc.groot.Groot/GetTxIDByCard', request, metadata || {}, this.methodInfoGetTxIDByCard);
    };
    return GrootClient;
}());
exports.GrootClient = GrootClient;
