// @ilt-core/ilt-form/lib/cjs/components/Inputs/FormSelect/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FormSelect = void 0;
var react_1 = __importStar(require("react"));
var react_hook_form_1 = require("react-hook-form");
var core_1 = require("@material-ui/core");
var icons_1 = require("../../icons");
var FormHelperText_1 = __importDefault(require("../../helper/FormHelperText"));
exports.FormSelect = react_1.default.forwardRef(function (props, ref) {
    var _a;
    var errors = react_hook_form_1.useFormContext().formState.errors;
    var name = props.name, label = props.label, placeholder = props.placeholder, value = props.value, 
    // helperText,
    disabled = props.disabled, options = props.options, popupIcon = props.popupIcon, onChange = props.onChange, onBlur = props.onBlur, onSync = props.onSync;
    var classes = useStyles();
    var _b = react_1.useState(false), open = _b[0], setOpen = _b[1];
    var handleClick = function () {
        setOpen(function (open) { return !open; });
    };
    var handleChange = function (event) {
        onChange(event);
        onSync();
    };
    var onClose = function () {
        setOpen(false);
    };
    var onOpen = function () {
        setOpen(true);
    };
    return (react_1.default.createElement(core_1.FormControl, { fullWidth: true, error: !!errors[name] },
        react_1.default.createElement(core_1.InputLabel, { shrink: true }, label),
        react_1.default.createElement(core_1.Select, { disabled: !!disabled, value: value || '', placeholder: placeholder, inputRef: ref, onBlur: onBlur, onChange: handleChange, open: open, onOpen: onOpen, onClose: onClose, endAdornment: react_1.default.createElement(core_1.IconButton, { onClick: handleClick, tabIndex: "-1" }, popupIcon || react_1.default.createElement(icons_1.ArrowDropDown, null)), IconComponent: function () { return react_1.default.createElement(react_1.default.Fragment, null); }, MenuProps: {
                anchorOrigin: {
                    vertical: 'bottom',
                    horizontal: 'left',
                },
                transformOrigin: {
                    vertical: -4,
                    horizontal: 'left',
                },
                getContentAnchorEl: null,
            }, displayEmpty: true, renderValue: value !== undefined
                ? undefined
                : function () { return react_1.default.createElement("div", { className: classes.placeholder }, placeholder); } }, options === null || options === void 0 ? void 0 : options.map(function (option) { return (react_1.default.createElement(core_1.MenuItem, { value: option.value, key: option.value }, option.label)); })),
        react_1.default.createElement(core_1.FormHelperText, null,
            react_1.default.createElement(FormHelperText_1.default, { message: (_a = errors[name]) === null || _a === void 0 ? void 0 : _a.message }))));
});
var useStyles = core_1.makeStyles({
    placeholder: {
        opacity: 0.4,
    },
});
