// @ilt-core/toast/build/hocs/withToast.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { ToastContainer } from 'react-toastify';
import styles from '../styles/styles.module.scss';
var configDefault = {
    hideProgressBar: true,
    closeButton: false,
    icon: false,
    containerId: styles.toast,
    className: styles.i_toast,
    toastClassName: styles.i_toast_item
};
var withToast = function (WrapperComponent, config) { return function (props) {
    var c = __assign(__assign({}, configDefault), config);
    return (React.createElement(React.Fragment, null,
        React.createElement(WrapperComponent, __assign({}, props)),
        React.createElement(ToastContainer, __assign({}, c))));
}; };
export default withToast;
