// @ilt-core/ilt-form/lib/esm/components/Inputs/FormCheckBoxGroup/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { FormControl, FormControlLabel, FormHelperText, makeStyles, } from '@material-ui/core';
import MyHelperText from '../../helper/FormHelperText';
import { useFormContext } from 'react-hook-form';
import { CheckboxGroup } from './CheckboxGroup';
export var FormCheckBoxGroup = function (props) {
    var _a;
    var name = props.name, label = props.label;
    var errors = useFormContext().formState.errors;
    var classes = useStyles();
    return (React.createElement(FormControl, { classes: { root: classes.MuiFormControlRoot } },
        React.createElement(FormControlLabel, { classes: { root: classes.MuiFormControlLabel }, label: label, control: React.createElement(CheckboxGroup, __assign({}, props)) }),
        React.createElement(FormHelperText, null,
            React.createElement(MyHelperText, { message: (_a = errors[name]) === null || _a === void 0 ? void 0 : _a.message }))));
};
var useStyles = makeStyles({
    MuiFormControlRoot: {
        marginTop: 0,
        width: '100%',
    },
    MuiFormControlLabel: {
        margin: 0,
    },
});
